src/loaders/helpers/apply-middlewares.ts

```diff
@@ -54,17 +54,20 @@ export function applyAfterAuthMiddleware(app: Router, middlewares: GetInjectable
 }
 
 function applyMiddleware(app: Router, middleware: GetInjectableOption<'middleware'>): void {
-	app.use(async (req: MedusaRequest | MedusaAuthenticatedRequest, res: Response, next: NextFunction) => {
-		const shouldHandle = middleware.routes.some((route) => {
-			return (
-				(route.method === 'all' || req.method.toLowerCase() === route.method.toLowerCase()) &&
-				(route.path === '*' || req.originalUrl === route.path)
-			);
-		});
-		if (shouldHandle) {
-			await new middleware.metatype().consume(req, res, next);
-			return;
+	const routes = middleware.routes.map((routeObj) => routeObj.path);
+	app.use(
+		routes,
+		async (routes, req: MedusaRequest | MedusaAuthenticatedRequest, res: Response, next: NextFunction) => {
```

Comment:
are you sure that the handler take `routes`? I don't find that in the doc [here](http://expressjs.com/en/4x/api.html#path-examples)

Reply:
Ah Sorry,  maybe I added mistakenly while formatting.
I 'll fix it in my next commit after testing. 
Thanks for adding integration tests so quickly.